decoder/impeghd_hoa_spatial_decoder.c

```diff
@@ -619,20 +628,29 @@ impeghd_hoa_spatial_decode_frame_side_info(pVOID handle, ia_hoa_frame_struct *ia
   }
 
   pstr_spatial_dec->scratch = ptr_coeff;
-  num_coeffs = (UWORD32)((enable_coeff - ptr_coeff) / sizeof(enable_coeff[0]));
+  num_coeffs = (UWORD32)(((size_t)enable_coeff - (size_t)ptr_coeff) / sizeof(enable_coeff[0]));
   frame_param->num_enable_coeff = num_coeffs;
   impeghd_hoa_spatial_sort_array_ascending(ptr_coeff, frame_param->amb_coeff_indices_to_enable,
                                            HOA_MAXIMUM_SET_SIZE);
+  while (frame_param->amb_coeff_indices_to_enable[count_enable] != -1) {
+    count_enable = count_enable + 1;
+  }
+  num_coeffs = count_enable;
```

Comment:
You can avoid num_coeffs = count_enable; statement here

Reply:
@shikhashetgeri-861 - It must be possible to remove line 631 and 632 as well, since the number of coefficients will now be computed entirely based on `amb_coeff_indices_to_enable` - buffer entries.